tools/restored-receipts-verifier/src/main.rs

```diff
@@ -108,6 +108,7 @@ fn main() -> Result<()> {
                 *block.header().random_value(),
                 false,
                 false, // because fix was not applied in for the blocks analyzed here
+                None,
```

Comment:
Another convention we seem to follow is to avoid having `Option` in input parameters list for APIs like that. Do we really need it?

Reply:
ah this one is hard to avoid. It was no argument in case of not `sandbox`, but the trouble of different signature under different feature was worse. So I move it to always have this argument, and it has to be an Option to differentiate with/without sandbox state patches.

Reply:
OK, although it correlate with my another advice of not using features. Generally this way of coding prevents tooling from being able to behave properly. But would really love to hear @matklad opinion here.

Reply:
The biggest smell I see here is not an addition of an argument per se, but rather the fact that this change requires us to update a dozen of unrelated tests. At least, my impression is that most of `None` come from tests. 

However, I also do wonder why do we add an extra parameter, rather than an extra method? That's, I'd expected this to work more like

```rust
diff --git a/chain/chain/src/chain.rs b/chain/chain/src/chain.rs
index 87b4eac7..719e3869 100644
--- a/chain/chain/src/chain.rs
+++ b/chain/chain/src/chain.rs
@@ -2911,13 +2911,13 @@ impl<'a> ChainUpdate<'a> {
                             *block.header().random_value(),
                             true,
                             is_first_block_with_chunk_of_version,
-                            #[cfg(feature = "sandbox")]
-                            self.states_to_patch.take(),
-                            #[cfg(not(feature = "sandbox"))]
-                            None,
                         )
                         .map_err(|e| ErrorKind::Other(e.to_string()))?;
 
+                    if let Some(states_to_patch) = self.states_to_patch {
+                        self.runtime_adapter.patch_state(states_to_patch)
+                    }
+
                     let (outcome_root, outcome_paths) =
                         ApplyTransactionResult::compute_outcomes_proof(&apply_result.outcomes);
 
```


I don't really understand the code in question though, so I am fine with the current impl.

Reply:
Hmm did a try this is not trivial to add, apply_state must be part of the runtime_adapter.apply_transaction, (calls runtime.apply) because this one returns the ApplyResult, which is most importantly `{trie_changes, state_changes, proof}`. If do patch_state after apply_transaction, then trie_changes & proof need to be recalculated, which were calculated as last step in runtime.apply. So it's still most neat to do patch_state as part of runtime.apply. So I'll keep this as is